modules/siddhi-extensions/event-table/src/main/java/org/wso2/siddhi/extension/eventtable/HazelcastEventTable.java

```diff
@@ -0,0 +1,308 @@
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.siddhi.extension.eventtable;
+
+import com.hazelcast.client.HazelcastClient;
+import com.hazelcast.client.config.ClientConfig;
+import com.hazelcast.config.Config;
+import com.hazelcast.core.Hazelcast;
+import com.hazelcast.core.HazelcastInstance;
+import org.apache.log4j.Logger;
+import org.wso2.siddhi.core.config.ExecutionPlanContext;
+import org.wso2.siddhi.core.event.ComplexEvent;
+import org.wso2.siddhi.core.event.ComplexEventChunk;
+import org.wso2.siddhi.core.event.MetaComplexEvent;
+import org.wso2.siddhi.core.event.stream.MetaStreamEvent;
+import org.wso2.siddhi.core.event.stream.StreamEvent;
+import org.wso2.siddhi.core.event.stream.StreamEventCloner;
+import org.wso2.siddhi.core.event.stream.StreamEventPool;
+import org.wso2.siddhi.core.event.stream.converter.ZeroStreamEventConverter;
+import org.wso2.siddhi.core.exception.OperationNotSupportedException;
+import org.wso2.siddhi.core.executor.VariableExpressionExecutor;
+import org.wso2.siddhi.core.table.EventTable;
+import org.wso2.siddhi.core.util.SiddhiConstants;
+import org.wso2.siddhi.core.util.collection.operator.Finder;
+import org.wso2.siddhi.core.util.collection.operator.Operator;
+import org.wso2.siddhi.extension.eventtable.hazelcast.HazelcastEventTableConstants;
+import org.wso2.siddhi.extension.eventtable.hazelcast.HazelcastOperatorParser;
+import org.wso2.siddhi.extension.eventtable.hazelcast.internal.ds.HazelcastEventTableServiceValueHolder;
+import org.wso2.siddhi.query.api.annotation.Annotation;
+import org.wso2.siddhi.query.api.definition.Attribute;
+import org.wso2.siddhi.query.api.definition.TableDefinition;
+import org.wso2.siddhi.query.api.expression.Expression;
+import org.wso2.siddhi.query.api.util.AnnotationHelper;
+
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentMap;
+
+/**
+ * Hazelcast event table implementation of SiddhiQL
+ */
+public class HazelcastEventTable implements EventTable {
+    private static final Logger logger = Logger.getLogger(HazelcastEventTable.class);
+    private final ZeroStreamEventConverter eventConverter = new ZeroStreamEventConverter();
+    private TableDefinition tableDefinition;
+    private ExecutionPlanContext executionPlanContext;
+    private StreamEventCloner streamEventCloner;
+    private StreamEventPool streamEventPool;
+    private HazelcastInstance hcInstance;
+    private ConcurrentMap<Object, StreamEvent> eventsMap = null;
```

Comment:
@tishan89 reasons;
1. Hazelcast Map (`IMap`) extends the interface `java.util.concurrent.ConcurrentMap`.
2. Since it is the distributed, might require concurrent modification of the Map from several threads without the need to block them.